src/tiffvisitor_int.cpp

```diff
@@ -473,7 +473,11 @@ namespace Exiv2 {
                                                       object->group());
         // skip decoding if decoderFct == 0
         if (decoderFct) {
-            EXV_CALL_MEMBER_FN(*this, decoderFct)(object);
+            if ( object->group() == nikonAf22Id ) {
+                EXV_CALL_MEMBER_FN(*this, decoderFct)(object);
```

Comment:
The true and false branch are exactly the same, aren't they?

Reply:
Dan.  You are correct and I "flattened" it last week.

There was a load of debugging code in the nikonAf22Id branch at one time and it was removed.  After I submitted the code I noticed this and simplified.